bgpd/bgp_route.c

```diff
@@ -4008,11 +4008,14 @@ bgp_static_update_safi (struct bgp *bgp, struct prefix *p,
 
   if ((safi == SAFI_EVPN) || (safi == SAFI_MPLS_VPN) || (safi == SAFI_ENCAP))
     {
-      if (bgp_static->igpnexthop.s_addr)
+      if (afi == AFI_IP)
         {
-          bgp_attr_extra_get (&attr)->mp_nexthop_global_in = bgp_static->igpnexthop;
+          if (!bgp_static->igpnexthop.s_addr)
+            attr.nexthop = bgp->router_id;
```

Comment:
I think this is not correct.

Take a look at this link: http://blog.ipspace.net/2011/08/bgp-next-hop-processing.html
_"When a router originates a BGP route configured with a network router configuration command or through route redistribution (redistribute router configuration command), it sets the BGP next hop to the IGP next hop (the same value you’d find in the IP routing table). BGP next hop is set to 0.0.0.0 for routes with unknown next hops – connected interfaces, static routes to null 0 or summary routes configured with aggregate-address router configuration command."_

I think the ideal fix would be to find out why `igpnexthop` is not being set anymore in any part of the code.